### ISSUE TITLE
Support special "validation" component_type for task csv-to-oscal-cd

### ISSUE THREAD

**Issue by degenaro** [IBM/compliance-trestle#1373]
## Issue description / feature objectives

The `validation` type of component has special requirements.

The following are ignored: (perhaps with warning?)

- Rule_Description
- Parameter_Id
- Parameter_Description
- Parameter_Value_Alternatives
- Parameter_Value_Default
- Profile_Source
- Profile_Description
- Control_Id_List

The following are required:

Check_Id
Check_Description

The following are optional:

Fetcher_Id
Fetcher_Description

## Caveats / Assumptions

The control implementation block is not generated.

## Completion Criteria

- Help is suitably updated
- New & existing test cases pass

**Comment by vikas-agarwal76** [comment]
Some additonal processing information -
  - Any other column present will be added as a property.
  - When converting CSV to OSCAL for a vaidation component, the Rule_Id values should be matched to a rule for a non "validation" type component to ensure that the rule exists. If not, then a warning message should be displayed and this rule should not be processed. A property "reference-component" with value as the name of the component that defined this rule should be added for this rule in the validation component (to specify the component for which this rule and check applies).
  - Since, the validation component only captures the rule to check mappings, it will not have any "control-implementation" in OSCAL.

The TRI - https://github.ibm.com/compliance-trestle-testing/shared-trestle-workspace/blob/cd-tri/TRIs/CD.md capures the details.

**Comment by degenaro** [comment]
Revised help.

```
degenaro:trestle.workspace$ trestle task csv-to-oscal-cd -i
trestle.core.commands.task:101 WARNING: Config file was not configured with the appropriate section for the task: "[task.csv-to-oscal-cd]"
Help information for csv-to-oscal-cd task.

Purpose: From csv produce OSCAL component_definition file.


Configuration flags sit under [task.csv-to-oscal-cd]:
  title                = (required) the component definition title.
  version              = (required) the component definition version.
  csv-file             = (required) the path of the csv file. [1st row are column headings; 2nd row are column descriptions; 3rd row and beyond is data]
  required columns:      Component_Title
                         Component_Description
                         Component_Type
                         Rule_Id
                         Rule_Description*
                         Profile_Source*
                         Profile_Description*
                         Control_Id_List*
                         Namespace
  optional columns:      Parameter_Id*
                         Parameter_Description*
                         Parameter_Value_Alternatives*
                         Parameter_Value_Default*
                         Check_Id+
                         Check_Description+
  output-dir           = (required) the path of the output directory for synthesized OSCAL .json files.
  component-definition = (optional) the path of the existing component-definition OSCAL .json file.
  class.column-name    = (optional) the class to associate with the specified column name, e.g. class.Rule_Id = scc_class
  output-overwrite     = (optional) true [default] or false; replace existing output when true.
  validate-controls    = (optional) on, warn, or off [default]; validate controls exist in resolved profile.

Notes: * column is ignored for validation component type
       + column is required for validation component type
```

**Comment by degenaro** [comment]
Fixed by #1431 .
